apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -52,27 +49,29 @@ impl<T: QueryPlanner> CacheResolver<QueryKey, QueryPlannerContent>
     for CachingQueryPlannerResolver<T>
 {
     async fn retrieve(&self, key: QueryKey) -> Result<QueryPlannerContent, CacheResolverError> {
-        self.delegate
-            .get(key.0, key.1, key.2)
-            .await
-            .map_err(|err| err.into())
+        self.delegate.get(key).await.map_err(|err| err.into())
     }
 }
 
 #[async_trait]
-impl<T: QueryPlanner> QueryPlanner for CachingQueryPlanner<T> {
-    async fn get(
-        &self,
-        query: String,
-        operation: Option<String>,
-        options: QueryPlanOptions,
-    ) -> PlanResult {
-        let key = (query, operation, options);
-        self.cm.get(key).await.map_err(|err| err.into())
+impl<T: QueryPlanner + Clone> QueryPlanner for CachingQueryPlanner<T> {
+    async fn get(&self, key: QueryKey) -> PlanResult {
+        let entry = self.cm.get(&key).await;
```

Comment:
Is it no longer possible to receive an error from a delegate? It feels like it should be, so I don't understand why that has changed. If a delegate returns an error for a key, how will we propagate that information to the client?

Note: I'm not 100% sure the original code handled this and so I may be over-thinking it here.

Reply:
`Planresult` can contain an error: `type PlanResult = Result<QueryPlannerContent, QueryPlannerError>;`
 
 That error should be stored in the cache because the query planner should always return the same error for the same query+schema+operation name. Maybe we should treat router-bridge error differently (ie when there's a communication error between Rust and V8), but right now we have no way to recover from such an error